geth/account/accounts.go

```diff
@@ -14,30 +13,37 @@ import (
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/status-im/status-go/extkeys"
-	"github.com/status-im/status-go/geth/common"
-	"github.com/status-im/status-go/geth/rpc"
 )
 
 // errors
 var (
+	ErrInvalidAccountAddressOrKey      = errors.New("cannot parse address or key to valid account address")
 	ErrAddressToAccountMappingFailure  = errors.New("cannot retrieve a valid account for a given address")
 	ErrAccountToKeyMappingFailure      = errors.New("cannot retrieve a valid key for a given account")
 	ErrWhisperIdentityInjectionFailure = errors.New("failed to inject identity into Whisper")
 	ErrWhisperClearIdentitiesFailure   = errors.New("failed to clear whisper identities")
 	ErrNoAccountSelected               = errors.New("no account has been selected, please login")
 	ErrInvalidMasterKeyCreated         = errors.New("can not create master extended key")
+	ErrInvalidAccountManager           = errors.New("could not retrieve account manager")
+	ErrAccountKeyStoreMissing          = errors.New("account key store is not set")
 )
 
+// ServiceProvider provides node and required services.
+type ServiceProvider interface {
```

Comment:
A bit unlucky with the generic name here. Sadly got no real better idea. Possibly only `Provider` so that it's used as `account.Provider`. But nothing important.

Reply:
Maybe if we split this interface into 2 smaller ones, it'll be cleaner and easier to name them?